v2/cmd/launch/main.go

```diff
@@ -31,13 +32,18 @@ var (
 	pipelineRunID     = flag.String("pipeline_run_id", "", "The current pipeline run ID.")
 	pipelineTaskID    = flag.String("pipeline_task_id", "", "The current pipeline task ID.")
 	pipelineRoot      = flag.String("pipeline_root", "", "The root output directory in which to store output artifacts.")
-	enableCaching     = flag.Bool(   "enable_caching", false, "Enable caching or not")
+	enableCaching     = flag.String(   "enable_caching", "false", "Enable caching or not")
 )
 
 func main() {
 	flag.Parse()
 	ctx := context.Background()
 
+	enableCachingBool, err := strconv.ParseBool(*enableCaching)
```

Comment:
for my education, what's the difference between this and flag.Bool? maybe comment the reason here?

Reply:
When using flag.Bool directly, the result of flags.args here have `true` in the array which will break parser logic https://github.com/kubeflow/pipelines/blob/c6cb8acf7a2b3a9361b02e059249433538293c9b/v2/component/launcher.go#L229 
<img width="635" alt="Screen Shot 2021-07-13 at 6 58 53 PM" src="https://user-images.githubusercontent.com/52932582/125548676-779b61e4-7d5d-46c0-82f1-f2fd55bf3fbf.png">

Reply:
Ohh, you are right. `flag.Bool` does not support `--enable_caching true` syntax,
see https://pkg.go.dev/flag#hdr-Command_line_flag_syntax

> -flag
-flag=x
-flag x  // non-boolean flags only

Reply:
@capri-xiyue can you include the ref in the comments too, I think that makes it clearer.
The root cause is that `-flag x` format is not supported for boolean flag, therefore the `true` item is not recognized and left in the args array.

Reply:
done